ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -245,19 +255,34 @@ protected BlockCreationResult createBlock(
               newProtocolSpec,
               pluginTransactionSelector,
               selectorsStateManager,
-              parentHeader);
+              parentHeader,
+              blockAccessListBuilder);
       transactionResults.logSelectionStats();
       timings.register("txsSelection");
       throwIfStopped();
 
+      final Optional<TransactionAccessList> postExecutionAccessList =
+          Optional.of(
+              new TransactionAccessList(transactionResults.getSelectedTransactions().size() + 1));
+
       final Optional<WithdrawalsProcessor> maybeWithdrawalsProcessor =
           newProtocolSpec.getWithdrawalsProcessor();
+
+      WorldUpdater worldUpdater = disposableWorldState.updater();
+      WorldUpdater postExecutionUpdater = worldUpdater.updater();
+      if (!(postExecutionUpdater instanceof StackedUpdater<?, ?>)) {
+        postExecutionUpdater = worldUpdater;
+      }
       final boolean withdrawalsCanBeProcessed =
           maybeWithdrawalsProcessor.isPresent() && maybeWithdrawals.isPresent();
       if (withdrawalsCanBeProcessed) {
         maybeWithdrawalsProcessor
             .get()
-            .processWithdrawals(maybeWithdrawals.get(), disposableWorldState.updater());
+            .processWithdrawals(
+                maybeWithdrawals.get(), postExecutionUpdater, postExecutionAccessList);
+        if (postExecutionUpdater instanceof StackedUpdater<?, ?>) {
+          worldUpdater.commit();
```

Comment:
instead of adding a new updater. we cannot not just remove the commit inside the processWithdrawals? I think like that we are adding more updater and the code will be more complex to follow